addons/account/models/account_account.py

```diff
@@ -621,15 +662,24 @@ def _get_most_frequent_accounts_for_partner(self, company_id, partner_id, move_t
             _kind, rhs_table, condition = query._joins['account_move_line__account_id']
             query._joins['account_move_line__account_id'] = (SQL("RIGHT JOIN"), rhs_table, condition)
 
-        from_clause, where_clause, params = query.get_sql()
-        self._cr.execute(f"""
-            SELECT account_move_line__account_id.id
-              FROM {from_clause}
-             WHERE {where_clause}
-          GROUP BY account_move_line__account_id.id
-          ORDER BY COUNT(account_move_line.id) DESC, account_move_line__account_id.code
-                   {f"LIMIT {limit:d}" if limit else ""}
-        """, params)
+        company = self.env['res.company'].browse(company_id)
+        code_sql = self.with_company(company)._field_to_sql('account_move_line__account_id', 'code', query)
```

Comment:
What does the `with_company` change? :thinking:

Reply:
My implementation of `_traverse_company_dependent_sql` uses the current company to figure out the domain on `ir.property` that corresponds to the current company's values.